### PR TITLE
fixes for gen2 CAS

### PR DESCRIPTION
**Pull request by Rohitrajak1807**

### CODE REVIEW

File: cmd/cluster/create.go [hunk 1]
```diff
@@ -133,7 +133,7 @@ func createClusterCommand() *cobra.Command {
 	command.Flags().StringVar(&argocdNs, "argocd-ns", "argocd", "the argocd namespace")
 	command.Flags().StringVar(&arlonNs, "arlon-ns", "arlon", "the arlon namespace")
 	command.Flags().StringVar(&arlonRepoUrl, "arlon-repo-url", "https://github.com/arlonproj/arlon.git", "the git repository url for arlon template")
-	command.Flags().StringVar(&arlonRepoRevision, "arlon-repo-revision", "v0.9.0", "the git revision for arlon template")
+	command.Flags().StringVar(&arlonRepoRevision, "arlon-repo-revision", "private/Rohitrajak1807/chc-gen2", "the git revision for arlon template")
```
Comment (bcle) [inline cmd/cluster/create.go:136]: Can you file a github issue to change this to v0.10.0 branch when it exists.

File: pkg/basecluster/prepare.go [hunk 2]
```diff
@@ -68,19 +74,31 @@ func Prepare(fileName string, validateOnly bool) (clusterName string, modifiedYa
...
 		unstr.SetNamespace("")
 		modified = true
 	}
+	if kind == "MachineDeployment" {
+		annotations := unstr.GetAnnotations()
+		if annotations == nil {
+			annotations = map[string]string{}
+		}
+
+		annotations[casMaxAnnotationMachineDeployments] = strconv.Itoa(casMax)
```
Comment (bcle) [inline pkg/basecluster/prepare.go:97]: I was thinking: if the user's manifest already has these annotations defined, shouldn't we leave them alone?

File: pkg/basecluster/prepare.go [hunk 3]
```diff
@@ -68,19 +74,31 @@ func Prepare(fileName string, validateOnly bool) (clusterName string, modifiedYa
 
 // -----------------------------------------------------------------------------
 
-func removeNamespaceThenEncode(obj runtime.Object, enc *yaml.Encoder) (modified bool, err error) {
+func removeNamespaceThenEncode(obj runtime.Object, enc *yaml.Encoder, casMax, casMin int) (modified bool, err error) {
```
Comment (bcle) [inline pkg/basecluster/prepare.go:77]: Now that this function does more than remove namespace, let's rename it to something like `performModsThenEncode()`, and then document here what changes the function makes.

File: pkg/basecluster/prepare.go [hunk 4]
```diff
@@ -68,19 +74,31 @@ func Prepare(fileName string, validateOnly bool) (clusterName string, modifiedYa
...
 	log := logpkg.GetLogger()
 	unstr := &unstructured.Unstructured{}
 	if err := scheme.Scheme.Convert(obj, unstr, nil); err != nil {
 		return false, fmt.Errorf("failed to convert object: %s", err)
 	}
+	kind := unstr.GetKind()
```
Comment (bcle) [inline pkg/basecluster/prepare.go:83]: Move to before line 91, where it's needed. Better yet: if this variable is only used once, let's not declare it, and use `if ustr.GetKind() == "MachineDeployment"`. If it gets used more than once in the future, we can reintroduce the variable.

**Review by bcle — CHANGES_REQUESTED** [review]

File: pkg/basecluster/prepare.go (hunk 2 above)
Comment (Rohitrajak1807) [reply pkg/basecluster/prepare.go:97]: will do

File: cmd/cluster/create.go (hunk 1 above)
Comment (Rohitrajak1807) [reply cmd/cluster/create.go:136]: https://github.com/arlonproj/arlon/issues/375

File: pkg/basecluster/prepare.go (hunk 3 above)
Comment (Rohitrajak1807) [reply pkg/basecluster/prepare.go:77]: done.
changed to `prepareCAPIManifestThenEncode`

File: pkg/basecluster/prepare.go (hunk 4 above)
Comment (Rohitrajak1807) [reply pkg/basecluster/prepare.go:83]: done

**Pull request merged**